der/src/tag.rs

```diff
@@ -433,6 +440,10 @@ impl Encode for Tag {
 
         let number = self.number().value();
 
+        // if writer.encoding_rules() == EncodingRules::Der {
```

Comment:
I would really prefer not to add support for encoding non-DER-conforming BER productions to this library. I'm reluctant to even add BER support but it's necessary to parse formats we support